src/test/clojure/clara/test_rules.clj

```diff
@@ -1183,75 +1183,85 @@
                                  [?t <- Temperature (= ?l location)]
                                  [:not [Cold (= temperature (:temperature ?t))]]]]])
 
-        s (mk-session [cold-not-match-temp] :cache false)
-        s-with-prior (mk-session [negation-with-prior-bindings] :cache false)
-        s-with-nested (mk-session [nested-negation-with-prior-bindings] :cache false)]
+        object-query (dsl/parse-query [] [[?o <- Object]])
 
-    (is (= [{}]
-           (-> s
-               (fire-rules)
-               (query cold-not-match-temp))))
+        s (mk-session [cold-not-match-temp object-query] :cache false)
+        s-with-prior (mk-session [negation-with-prior-bindings object-query] :cache false)
+        s-with-nested (mk-session [nested-negation-with-prior-bindings object-query] :cache false)
+
+        no-system-types? (fn [session]
```

Comment:
It isn't clear to me why this check has been added to all of these test cases.  It looks like revealing an impl detail unnecessarily in the test, but I'm guessing I'm missing something?

If there is a meaningful reason to do this, a comment above this line may be helpful to make that clear for future reading.